### ISSUE TITLE
AddCallback() being ignored within table when last in column

### ISSUE THREAD

**Issue by hoffstadt** [ocornut/imgui#4843]
**Version/Branch of Dear ImGui:**

Version: 1.83
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_dx11.cpp + imgui_impl_win32.cpp
Compiler: MSVC
Operating System: Windows 10

**My Issue/Question:**
Inside a table, calling `ImGui::AddCallback(...)` as the last item in a column has no effect.

If you use the __ImGuiTableFlags_NoClip__ flag, it works.

**Standalone, minimal, complete and verifiable example:**

```cpp
// includes

void callback0(const ImDrawList* parent_list, const ImDrawCmd* cmd)
{
    std::cout << "Callback 0" << std::endl;
}

void callback1(const ImDrawList* parent_list, const ImDrawCmd* cmd)
{
    std::cout << "Callback 1" << std::endl;
}

int main()
{
    // boilerplate

    while (true)
    {
        // boilerplate

        ImGui_ImplDX11_NewFrame();
        ImGui_ImplWin32_NewFrame();
        ImGui::NewFrame();

        // boilerplate

        ImGui::Begin("Main Window");
        ImGui::GetWindowDrawList()->AddCallback(callback0, nullptr);
        ImGui::Button("Press me");
        ImGui::GetWindowDrawList()->AddCallback(callback1, nullptr);
        ImGui::End();

        static ImGuiTableFlags tableflags =
            ImGuiTableFlags_Borders |
            ImGuiTableFlags_Resizable |
            ImGuiTableFlags_SizingStretchProp |
            ImGuiTableFlags_NoHostExtendX;

        if (ImGui::BeginTable("Table ID", 1, tableflags))
        {
            ImGui::TableNextColumn();
            ImGui::TableSetColumnIndex(0);
            ImGui::GetWindowDrawList()->AddCallback(callback0, nullptr);
            ImGui::Button("B0");
            ImGui::GetWindowDrawList()->AddCallback(callback1, nullptr);
            ImGui::EndTable();
        }

        ImGui::Render();
        ImGui_ImplDX11_RenderDrawData(ImGui::GetDrawData());

        // boilerplate
    }

    // boilerplate
}
```

**Comment by ocornut** [comment]
Thank you. Fixed by your PR #4844 see comments there for details.

**Comment by ocornut** [comment]
Facepalm of the year for neither of us not noticing the bug here. Fixing soon!

**Comment by ocornut** [comment]
The PR was wrong, we didn't noticed as other flags which led to additional drawing (BordersH, RowBG) would strip that empty draw command in other places. Discovered by @rokups.
Pushed fix f58bd817e2998489bace1b2dff49884eac790efb + amended our tests.
